### ISSUE TITLE
Make the stats history data to support groupIds query params and use new API to render the stats history UI

### ISSUE THREAD

**Issue by LieutenantRoger** [topcoder-platform/community-app#4137]
1. Community-app - Allowing seeing stats history for topgear / other communities. Was disable during last release.
- code access: 
https://github.com/topcoder-platform/community-app/tree/feature-stats-history

2. Community-app - Change Stats History web service call to fetch data by passing groupsId, as we did for member stats, and use the retrieved stats from new API to render the UI.

Please make sure both public history page and topgear history page are working correctly.

- code access:
https://github.com/topcoder-platform/community-app/tree/feature-stats-history
https://github.com/topcoder-platform/topcoder-react-lib/tree/feature-stats-history

**Comment by LieutenantRoger** [comment]
Also please only test `topgear` group for now.

**Comment by LieutenantRoger** [comment]
FYI @urwithat @Oanh-and-only-Oanh

**Comment by SathyaJayabal** [comment]
@LieutenantRoger , was able to view both the public and private stats history page for the user standlove. There are a few inconsistencies in the data like the below
1. Even though the ratings are different for the design, architecture and assembly tracks(for public and private history), the challenge details are all the same
<img width="1440" alt="Screenshot 2020-03-30 at 1 27 46 PM" src="https://user-images.githubusercontent.com/58783823/77888720-4ceee780-728a-11ea-8184-619cb1814299.png">

2. The rating history graph for topgear for the design track does not match the rating
<img width="1424" alt="Screenshot 2020-03-30 at 1 23 45 PM" src="https://user-images.githubusercontent.com/58783823/77888731-51b39b80-728a-11ea-9491-d3facdbff1fc.png">
3. Even though the. user has the same data. for SRMs in both the public and private history pages, the rating history graph is not displayed for topgear.
<img width="1440" alt="Screenshot 2020-03-30 at 1 30 37 PM" src="https://user-images.githubusercontent.com/58783823/77889051-dacad280-728a-11ea-995d-ecab7fcc8c61.png">
<img width="1440" alt="Screenshot 2020-03-30 at 1 30 43 PM" src="https://user-images.githubusercontent.com/58783823/77889073-e28a7700-728a-11ea-94c2-638d39138039.png">
4. Rating distribution graph for private history of  assembly track does not show the users rating.

<img width="1440" alt="Screenshot 2020-03-30 at 1 32 37 PM" src="https://user-images.githubusercontent.com/58783823/77889180-19608d00-728b-11ea-8e8e-a9d14cc61326.png">

**Comment by LieutenantRoger** [comment]
Hi @SathyaJayabal 

As have discussed, this is the expected behavior due to the testing data preparation.   

Btw, I have re-deployed the feature using topcoder-react-lib#v1000.14.0 to dev env for QA.

**Comment by SathyaJayabal** [comment]
@LieutenantRoger for the. topcoder stats history page(public) there should not abe any group id passed right?

**Comment by SathyaJayabal** [comment]
@LieutenantRoger for the call to stats history page for topgear, there are. two group ids passed. Topgear group is "20000000", why is the other id passed ?
<img width="1440" alt="Screenshot 2020-03-30 at 5 04 38 PM" src="https://user-images.githubusercontent.com/58783823/77908237-bd0c6600-72a8-11ea-8499-b803698857c9.png">

**Comment by LieutenantRoger** [comment]
It's expected as well, please refer the `stats` call for reference when you are checking `history` call.

**Comment by LieutenantRoger** [comment]
> @LieutenantRoger for the. topcoder stats history page(public) there should not abe any group id passed right?

Correct, when there is no group id provided in the request,  the history data with groupId=10 will be returned by default.

**Comment by SathyaJayabal** [comment]
@LieutenantRoger , I am. passing this one as per clarifications above. The public and topgear stats history pages are displayed for the user standlove. The group id is passed in the history call.

**Comment by SathyaJayabal** [comment]
verified on dev(develop branch)
<img width="1440" alt="Screenshot 2020-03-31 at 2 20 32 PM" src="https://user-images.githubusercontent.com/58783823/78006786-0a491000-735b-11ea-8672-a11e836a73c3.png">
<img width="1440" alt="Screenshot 2020-03-31 at 2 20 43 PM" src="https://user-images.githubusercontent.com/58783823/78006819-159c3b80-735b-11ea-98c6-390d9e682a83.png">
<img width="1440" alt="Screenshot 2020-03-31 at 2 21 19 PM" src="https://user-images.githubusercontent.com/58783823/78006834-1af98600-735b-11ea-89fd-737635f0635f.png">
<img width="1440" alt="Screenshot 2020-03-31 at 2 21 42 PM" src="https://user-images.githubusercontent.com/58783823/78006856-22b92a80-735b-11ea-8068-5465d07a2bee.png">

**Comment by LieutenantRoger** [comment]
Thanks @SathyaJayabal

**Comment by LieutenantRoger** [comment]
@lakshmiathreya  please close if the ticket is resolved. It's been long time now.

**Comment by SathyaJayabal** [comment]
@LieutenantRoger ,the stats data is still not visible for top gear users in production. @lakshmiathreya can u verify this please. Clicking on any user from the topgear leaderboard should show the profile page with stats.

**Comment by SathyaJayabal** [comment]
@LieutenantRoger the below user must have stats data in topgear but the stats are not showing up
<img width="1440" alt="Screenshot 2020-04-14 at 10 23 24 PM" src="https://user-images.githubusercontent.com/58783823/79252016-97817e00-7e9e-11ea-8e20-38b231a9812c.png">

**Comment by LieutenantRoger** [comment]
@SathyaJayabal  got it.  I just remembered the production data under topgear are still under processing, I  think this is the reason that no data is returned, we can keep the ticket as open now, thanks.

**Comment by sushilshinde** [comment]
@LieutenantRoger need pr for this

**Comment by SathyaJayabal** [comment]
marking this as hold until backend data is ready for topgear stats history on prod
cc @LieutenantRoger @Oanh-and-only-Oanh

**Comment by lakshmiathreya** [comment]
@Oanh-and-only-Oanh I am reopening this - it appears like Code was fixed but backend data was not available. Will leave this open until we see Stats data fully. cc @urwithat @SathyaJayabal
